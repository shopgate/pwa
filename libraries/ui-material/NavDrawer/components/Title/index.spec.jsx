import React from 'react';
import { mount } from 'enzyme';
import NavDrawerTitle from './index';

describe('<NavDrawerTitle />', () => {
  it('should render when a text is passed within the props', () => {
    const title = 'Title';
    const wrapper = mount(<NavDrawerTitle text={title} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual(title);
    expect(wrapper.find('Translate').prop('string')).toEqual(title);
  });

  it('should not render when no text is passed within the props', () => {
    const wrapper = mount(<NavDrawerTitle />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toBe(true);
  });
});
