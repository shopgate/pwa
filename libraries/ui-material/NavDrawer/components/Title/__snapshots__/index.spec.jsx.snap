// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavDrawerTitle /> should not render when no text is passed within the props 1`] = `
<NavDrawerTitle
  text=""
/>
`;

exports[`<NavDrawerTitle /> should render when a text is passed within the props 1`] = `
<NavDrawerTitle
  text="Title"
>
  <div
    className={
      Object {
        "data-css-o38fs9": "",
      }
    }
  >
    <Translate
      className=""
      params={Object {}}
      string="Title"
    >
      <span
        className=""
      >
        Title
      </span>
    </Translate>
  </div>
</NavDrawerTitle>
`;
